conftest.py

```diff
@@ -83,7 +85,32 @@ def pytest_addoption(parser):
              "Set it to 'auto' to let the fixtures auto-detect available disks."
     )
 
+def check_xo_cli_version():
+    try:
+        output = subprocess.check_output(['xo-cli', '--help'])
+    except FileNotFoundError:
+        pytest.exit("Could not find xo-cli in path")
+    lines = output.splitlines()
+    for line in lines:
+        line = line.decode('utf-8')
+        if not line.startswith("xo-cli "):
+            continue
+        match = re.match(rf"{re.escape('xo-cli')}\s+v(\d+)\.(\d+)\.(\d+)", line)
+
+        if match is None:
+            continue
+
+        major, minor, patch = map(int, match.groups())
+        if major <= 0 and minor < 17:
+            pytest.exit(f"Expected xo-cli >= v0.17.0, found v{major}.{minor}.{patch}".format(), 1)
+        else:
+            return
+
+    pytest.exit("Could not find xo-cli version", 1)
```

Comment:
I agree with the logic behind your suggestion, but I don't really see how to do it.

I assume the `else:` in your suggestion in meant to be for `if match:` which is not right we want to iterate on each line of the `xo-cli --help` because in some older version there was an empty line after the version, otherwise I would have used `output.splitline()[-1]` for only the last line.

For this reason we cannot know if we finished processing inside the loop and we can get out of the loop without having a match, therefore we would need to check if `major`, `minor`, `patch` are defined before checking them.

In order to achieve what  you suggest the only way I see would be something like:

```python
    for line in output.splitlines():
        line = line.decode('utf-8')
        if not line.startswith("xo-cli "):
            continue
        match = re.match(rf"{re.escape('xo-cli')}\s+v(\d+)\.(\d+)\.(\d+)", line)

        if match:
            major, minor, patch = map(int, match.groups())
            break

    if 'major' in locals() and 'minor'in locals():
        if major <= 0 and minor < 17:
            pytest.exit(f"Expected xo-cli >= v0.17.0, found v{major}.{minor}.{patch}".format(), 1)
    else:
        pytest.exit("Could not find xo-cli version", 1)
```

I'm not against it, just wondering if it in indeed more readable in the end, let me know what you think.

Reply:
No, the `else:` is for the `for:` loop, only matching if the version was not found in any output line.  I tried to have the indent correct, but the missing context likely does not help :)

Reply:
Ok, I wasn't even aware of the for else construct in python :sweat_smile: 

In any case, as discussed in the generic comments with Samuel, we likely won't have a way to implement this check without creating issues for now, so I guess we can close your 2 comments on this function if that's fine with you.